packages/next/client/future/image.tsx

```diff
@@ -380,34 +412,71 @@ export default function Image({
       // instead console.error(ref) during mount.
       unoptimized = true
     } else {
-      if (typeof widthInt === 'undefined') {
-        throw new Error(
-          `Image with src "${src}" is missing required "width" property.`
-        )
-      } else if (isNaN(widthInt)) {
-        throw new Error(
-          `Image with src "${src}" has invalid "width" property. Expected a numeric value in pixels but received "${width}".`
-        )
-      }
-      if (typeof heightInt === 'undefined') {
-        throw new Error(
-          `Image with src "${src}" is missing required "height" property.`
-        )
-      } else if (isNaN(heightInt)) {
-        throw new Error(
-          `Image with src "${src}" has invalid "height" property. Expected a numeric value in pixels but received "${height}".`
-        )
-      }
-      if (!VALID_LOADING_VALUES.includes(loading)) {
-        throw new Error(
-          `Image with src "${src}" has invalid "loading" property. Provided "${loading}" should be one of ${VALID_LOADING_VALUES.map(
-            String
-          ).join(',')}.`
-        )
+      if (fill) {
+        if (width) {
+          throw new Error(
+            `Image with src "${src}" has both "width" and "fill" properties. Images in fill mode should not specify width or height.`
+          )
+        }
+        if (height) {
+          throw new Error(
+            `Image with src "${src}" has both "height" and "fill" properties. Images in fill mode should not specify height or width.`
+          )
+        }
+        if (style?.position) {
```

Comment:
Yeah, it's going to be very easy to break the fill mode no-layout-shift guarantee by targeting the image with CSS, and I don't think there's anything we can really do about it. We COULD do a runtime check on the image's computed CSS position attribute, but I'm not sure we want to do that, as there are some cases where the image doesn't need position:absolute (such as if the containing element has an assigned height and width) so it wouldn't technically be wrong to change it.

I'm kind of on the fence about whether we should keep this check here, since it is one place where we have visibility into how the user is applying styles to the image, and we could at least warn them off from making certain basic mistakes.